packages/jest-test-sequencer/src/index.ts

```diff
@@ -66,24 +74,87 @@ export default class TestSequencer {
   }
 
   /**
-   * Sorting tests is very important because it has a great impact on the
-   * user-perceived responsiveness and speed of the test run.
+   * Select tests for shard requested via --shard=shardIndex/shardCount
+   * Sharding is applied before sorting
    *
-   * If such information is on cache, tests are sorted based on:
-   * -> Has it failed during the last run ?
-   * Since it's important to provide the most expected feedback as quickly
-   * as possible.
-   * -> How long it took to run ?
-   * Because running long tests first is an effort to minimize worker idle
-   * time at the end of a long test run.
-   * And if that information is not available they are sorted based on file size
-   * since big test files usually take longer to complete.
+   * @param tests All tests
+   * @param options shardIndex and shardIndex to select
    *
-   * Note that a possible improvement would be to analyse other information
-   * from the file other than its size.
+   * @example
+   * ```typescript
+   * class CustomSequencer extends Sequencer {
+   *  shard(tests, { shardIndex, shardCount }) {
+   *    const shardSize = Math.ceil(tests.length / options.shardCount);
+   *    const shardStart = shardSize * (options.shardIndex - 1);
+   *    const shardEnd = shardSize * options.shardIndex;
+   *    return [...tests]
+   *     .sort((a, b) => (a.path > b.path ? 1 : -1))
+   *     .slice(shardStart, shardEnd);
+   *  }
+   * }
+   * ```
+   */
+  shard(
+    tests: Array<Test>,
+    options: ShardOptions,
+  ): Array<Test> | Promise<Array<Test>> {
+    const shardSize = Math.ceil(tests.length / options.shardCount);
+    const shardStart = shardSize * (options.shardIndex - 1);
+    const shardEnd = shardSize * options.shardIndex;
+
+    return tests
+      .map(test => {
+        const relativeTestPath = path.relative(
+          test.context.config.rootDir,
+          test.path,
+        );
```

Comment:
a change in the logic to make the sharding stable between machines with different paths (`test.path` is an absolute path, so the hash differs between machines). Not really an issue except that we test the sharding in this repo, so we need to be stable 🙂